cpp/src/arrow/python/pandas_convert.cc

```diff
@@ -362,23 +362,25 @@ class PandasConverter {
   Status Visit(const Time32Type& type) { return VisitNative<Int32Type>(); }
   Status Visit(const Time64Type& type) { return VisitNative<Int64Type>(); }
 
-  Status Visit(const NullType& type) { return Status::NotImplemented("null"); }
+  template<typename T>
+  Status TypeNotImplemented(const T& type) {
+    std::stringstream ss;
+    ss << "PandasConverter doesn't implement <" << type.name() << "> conversion. ";
```

Comment:
You could pass `type.ToString()` to `TypeNotImplemented` and skip the complexity of the template (which need not be in the class scope)